subsys/usb/class/Kconfig.cdc

```diff
@@ -1,15 +1,26 @@
 # Copyright (c) 2016 Wind River Systems, Inc.
 # SPDX-License-Identifier: Apache-2.0
 
-menuconfig USB_CDC_ACM
-	bool "USB CDC ACM Device Class support"
+DT_COMPAT_ZEPHYR_CDC_ACM_UART := zephyr,cdc-acm-uart
+
+menu "USB CDC ACM Class support"
+
+config USB_CDC_ACM_HAS_DT_COMPAT
+	bool
+	default $(dt_compat_enabled,$(DT_COMPAT_ZEPHYR_CDC_ACM_UART))
+	imply USB_CDC_ACM
+	help
+	  Helper symbol to select USB_CDC_ACM if compatible node is enabled.
```

Comment:
I thought its better to have each class as menuconfig.
What is the advantage of having this hidden option vs having:

``` patch
diff --git a/subsys/usb/class/Kconfig.cdc b/subsys/usb/class/Kconfig.cdc
index bc9e169afa..f7e9d96465 100644
--- a/subsys/usb/class/Kconfig.cdc
+++ b/subsys/usb/class/Kconfig.cdc
@@ -3,21 +3,13 @@
 
 DT_COMPAT_ZEPHYR_CDC_ACM_UART := zephyr,cdc-acm-uart
 
-menu "USB CDC ACM Class support"
-
-config USB_CDC_ACM_HAS_DT_COMPAT
-	bool
-	default $(dt_compat_enabled,$(DT_COMPAT_ZEPHYR_CDC_ACM_UART))
-	imply USB_CDC_ACM
-	help
-	  Helper symbol to select USB_CDC_ACM if compatible node is enabled.
-
-config USB_CDC_ACM
+menuconfig USB_CDC_ACM
 	bool "Enable USB CDC ACM Class support"
 	select SERIAL_HAS_DRIVER
 	select SERIAL_SUPPORT_INTERRUPT
 	select RING_BUFFER
 	select UART_INTERRUPT_DRIVEN
+	default $(dt_compat_enabled,$(DT_COMPAT_ZEPHYR_CDC_ACM_UART))
 	help
 	  Enable USB CDC ACM Class support.
 	  Default UART device name is "CDC_ACM_0".
@@ -37,7 +29,7 @@ config USB_CDC_ACM_DEVICE_NAME
 	  Device name template for the CDC ACM Devices. First device would
 	  have name $(USB_CDC_ACM_DEVICE_NAME)_0, etc.
 
-if !USB_CDC_ACM_HAS_DT_COMPAT
+if !$(dt_compat_enabled,$(DT_COMPAT_ZEPHYR_CDC_ACM_UART))
 module = USB_CDC_ACM
 default-count = 1
 source "subsys/usb/class/Kconfig.template.composite_device_number"
@@ -77,5 +69,3 @@ module-str = usb cdc acm
 source "subsys/logging/Kconfig.template.log_config"
 
 endif # USB_CDC_ACM
-
-endmenu
```

The only thing that's different is not the best description for why is `USB_CDC_ACM` y selected, but is that really an issue?

Reply:
> I thought its better to have each class as menuconfig.
> What is the advantage of having this hidden option vs having:

It still has a menu because of `menu "USB CDC ACM Class support"`.
It is for debugging, USB_CDC_ACM_HAS_DT_COMPAT allows to see what _y-selecting this symbol_ or _y-implying_, here USB_CDC_ACM.

Reply:
There are many examples already in the tree that does the same without helper option. I think the solution i proposed is much more compact and elegant. I believe zephyr users are mature enough to understand why this is `y` implied by looking into the Kconfig directly. To clarify it you could add some note in help
```
 	help
 	  Enable USB CDC ACM Class support.
 	  Default UART device name is "CDC_ACM_0".
 	  This option is automatically enabled if USB CDC ACM
 	  is chosen as zephyr console in DTS.
```

Menuconfig option could be selected, menu requires to go one step deeper into configuration gui. I wouldn't change that.

Reply:
I disagree.

Reply:
USB_CDC_ACM_HAS_DT_COMPAT is removed in the commit "usb: cdc_acm: configure CDC ACM UART instances using devicetree only"